### PR TITLE
gluon-ts 0.15.1 

### PR DESCRIPTION
**Pull request by msentissi**
gluon-ts 0.15.1 

**Destination channel:** Defaults

### Links

- [PKG-6354]
- dev_url:        https://github.com/awslabs/gluonts/tree/v0.15.1
- conda_forge:    https://github.com/conda-forge/gluonts-feedstock
- pypi:           https://pypi.org/project/gluonts/0.15.1
- pypi inspector: https://inspector.pypi.io/project/gluonts/0.15.1

### Explanation of changes:

- Upgrade to v0.15.1

[PKG-6354]: https://anaconda.atlassian.net/browse/PKG-6354?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ requirements:
     # Capping `lightning` does not cap `pytorch_lightning`, so we cap manually
     - pytorch-lightning >=2.0,<2.2
     - scipy >=1.7.3,<1.8  # [py<38]
```
Comment (ELundby45) [inline recipe/meta.yaml:47]: I think this could be fixed too?

```suggestion
    - scipy >=1.7.3,<2  # [py<38]
```

**Review by ELundby45 — APPROVED** [review]

**Review by skupr-anaconda — APPROVED** [review]
LGTM but Eric's comment

File: recipe/meta.yaml (hunk 1 above)
Comment (msentissi) [reply recipe/meta.yaml:47]: My reasoning for this was that `~=1.7.3` is equivalent to `>=1.7.3, ==1.7.*` such that `1.8` would be excluded. I based this on this [document](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release). I'm not certain it's the right source to use though.

Comment (ELundby45) [reply recipe/meta.yaml:47]: You are correct, I was mistaken on how the `~=` works. Sorry!

**Comment by msentissi** [comment]
@ELundby45 @skupr-anaconda I've had to make some additional changes to the recipe: this was being built as a dependency for `autogluon` which requires `pytorch-lightning >=2.2,<2.4`. The "core" `gluonts` doesn't depend on `pytorch-lightning`, but we build it with the `torch` extra included, which entails `pytorch-lightning >=2.0,<2.2`. For this reason, I propose to relax the upperbound in this extra-dependency and add testing for `autogluon[torch]` to verify this doesn't break anything obvious.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -36,17 +42,38 @@ requirements:
...
+    - pytest >7.0
+    - pytest-cov >4.0,<5.0
+    - pytest-timeout >2.0, <3.0
+    - pytest-xdist >3.0, <4.0
+    - pytest-rerunfailures >=12.0, <14.0
+    - ujson
+    - orjson
+    - requests
+    - holidays >=0.9,<1
+    - matplotlib >=3.6,<4
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - matplotlib-base >=3.6,<4
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -36,17 +42,38 @@ requirements:
...
+  source_files:
+    - test/torch
   imports:
     - gluonts
   requires:
     - pip
+    - pytest
+    - pandas >=1.1
+    - pytest >7.0
+    - pytest-cov >4.0,<5.0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
```
pytest-cov  is redundant

**Pull request merged**